### PR TITLE
dovecot: 2.3.13 -> 2.3.14, dovecot_pigeonhole: 0.5.13 -> 0.5.14

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
https://github.com/dovecot/core/blob/2.3.14/NEWS

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/mail/dovecot/default.nix [hunk 1]
```diff
@@ -22,7 +22,7 @@ stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "https://dovecot.org/releases/2.3/${pname}-${version}.tar.gz";
```
Comment (drewrisinger) [inline pkgs/servers/mail/dovecot/default.nix:24]:
```suggestion
    url = "https://dovecot.org/releases/${lib.versions.majorMinor version}/${pname}-${version}.tar.gz";
```
https://github.com/NixOS/nixpkgs/blob/102eb68ceecbbd32ab1906a53ef5a7269dc9794a/lib/versions.nix#L38-L47

**Review by drewrisinger — APPROVED** [review]
* [x] Diff LGTM
* [x] Commits LGTM
* [x] Builds via ``nix-review`` on ``x86-64-linux``:

```
3 packages built:
dovecot dovecot_fts_xapian dovecot_pigeonhole
```

File: pkgs/servers/mail/dovecot/default.nix (hunk 1 above)
Comment (ajs124) [reply pkgs/servers/mail/dovecot/default.nix]: I don't except them to release 2.4 anytime soon, but why not. Also soft-coded it in pigeonhole.

**Comment by drewrisinger** [comment]
@ajs124 LGTM, except usually we prefer to have only one file touched per commit (the last soft-coded version touched two)

**Comment by ajs124** [comment]
> usually we prefer to have only one file touched per commit

we do?

**Comment by drewrisinger** [comment]
> > usually we prefer to have only one file touched per commit
> 
> we do?

That's my best interpretation from previous commits that I've seen, and https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes. There's obvious exceptions for e.g. changing ``all-packages.nix`` and the ``/package_name/default.nix`` files, or treewide changes.

**Pull request merged**